gpflow/models/sgpr.py

```diff
@@ -26,37 +26,68 @@
 from ..mean_functions import Zero, MeanFunction
 
 
-class SGPRUpperMixin(GPModel):
+class SGPRBase(GPModel):
+    """
+    Common base class for SGPR and GPRFITC that provides the common __init__
+    and upper_bound() methods.
     """
-    Upper bound for the GP regression marginal likelihood.
-    It is implemented here as a Mixin class which works with SGPR and GPRFITC.
-    Note that the same inducing points are used for calculating the upper bound,
-    as are used for computing the likelihood approximation. This may not lead to
-    the best upper bound. The upper bound can be tightened by optimising Z, just
-    as just like the lower bound. This is especially important in FITC, as FITC
-    is known to produce poor inducing point locations. An optimisable upper bound
-    can be found in https://github.com/markvdw/gp_upper.
 
-    The key reference is
+    def __init__(self,
+                 data: Data,
+                 kernel: Kernel,
+                 mean_function: Optional[MeanFunction] = None,
+                 inducing_variable: Optional[InducingPoints] = None,
+                 num_latent: Optional[int] = None,
+                 noise_variance: float = 1.0,
+                 ):
+        """
+        X is a data matrix, size [N, D]
+        Y is a data matrix, size [N, R]
+        Z is a matrix of pseudo inputs, size [M, D]
+        kernel, mean_function are appropriate GPflow objects
 
-    ::
+        This method only works with a Gaussian likelihood.
```

Comment:
i suggest
```suggestion
        Assumes a Gaussian likelihood.
```
that is, unless the args needs to be compatible with a Gaussian likelihood